workflow/src/main/java/bio/terra/tanagra/workflow/CopyBigQueryDataset.java

```diff
@@ -0,0 +1,152 @@
+package bio.terra.tanagra.workflow;
+
+import bio.terra.tanagra.proto.underlay.ColumnId;
+import bio.terra.tanagra.proto.underlay.Dataset;
+import bio.terra.tanagra.proto.underlay.EntityMapping;
+import bio.terra.tanagra.proto.underlay.Table;
+import bio.terra.tanagra.proto.underlay.Underlay;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.sql.DataSource;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.io.gcp.bigquery.BigQueryOptions;
+import org.apache.beam.sdk.io.jdbc.JdbcIO;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.transforms.SerializableFunction;
+import org.apache.commons.text.StringSubstitutor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** A batch Apache Beam pipeline for copying data from BigQuery to CloudSQL Postgres instances. */
+public class CopyBigQueryDataset {
```

Comment:
nit: should this pipeline name include the destination? e.g. `CopyBigQueryDatasetToPostgres`, or is the intent to eventually have the destination be parameterizable?

Reply:
Being specific sounds good. We can rename if it does parameterize.